lib/ansible/plugins/inventory/now.py

```diff
@@ -0,0 +1,215 @@
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+        proxy = self.get_option('proxy')
+
+        # build url
+        self.url = "https://%s/%s" % (self.get_option('instance'), path)
+        url = self.url
+        results = []
+
+        if not self.update_cache:
+            try:
+                results = self._cache[self.cache_key][self.url]
+            except KeyError:
+                pass
+
+        if not results:
+            if self.cache_key not in self._cache:
+                self._cache[self.cache_key] = {self.url: ''}
+
+            session = requests.Session()
+
+            while url:
+                # perform REST operation, accumulating page results
+                response = session.get(url,
+                                       auth=auth,
+                                       headers=headers,
+                                       proxies={
+                                           'http': proxy,
+                                           'https': proxy
+                                       })
+                if response.status_code != 200:
+                    raise AnsibleError("http error (%s): %s" %
+                                       (response.status_code, response.text))
+                results += response.json()['result']
+                next_link = response.links.get('next', {})
+                url = next_link.get('url', None)
+
+            self._cache[self.cache_key] = {self.url: results}
+
+        results = {'result': results}
+        return results
+
+    def parse(self, inventory, loader, path,
+              cache=True):  # Plugin interface (2)
+        super(InventoryModule, self).parse(inventory, loader, path)
+
+        if not HAS_REQUESTS:
+            raise AnsibleParserError('Please install "requests" Python module as this is required'
+                                     ' for ServiceNow dynamic inventory plugin.')
+
+        self._read_config_data(path)
+        self.cache_key = self.get_cache_key(path)
+
+        self.use_cache = self.get_option('cache') and cache
+        self.update_cache = self.get_option('cache') and not cache
+
+        selection = self.get_option('selection_order')
+        fields = self.get_option('fields')
+        table = self.get_option('table')
+        filter_results = self.get_option('filter_results')
+
+        options = "?sysparm_exclude_reference_link=true&sysparm_display_value=true"
+
+        path = '/api/now/table/' + table + options + \
+            "&sysparm_fields=" + ','.join(fields) + \
+            "&sysparm_query=" + filter_results
+
+        content = self.invoke('GET', path, None)
+        strict = self.get_option('strict')
+
+        for record in content['result']:
+
+            target = None
+
+            for k in selection:
+                if k in record:
+                    if record[k] != '':
+                        target = record[k]
```

Comment:
Should the first value found in the selection win, or the last one? I'm wondering if there should be a `break` after this line if target is not None.

Reply:
The behavior of the inventory script was last match wins but I think first match wins is more intuitive for setting the selection_order variable.